src/main/java/org/example/tms/service/impl/TaskServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class TaskServiceImpl implements TaskService {
+    private final TaskMapper taskMapper;
+    private final UserService userService;
+    private final TaskRepository taskRepository;
+    private final PermissionValidator permissionValidator;
+
+    @Override
+    @ExecutionTime
+    @Transactional(readOnly = true)
+    public Task getTaskEntityById(UUID id) {
+        return taskRepository.findById(id)
+                .orElseThrow(() -> new TaskNotFoundException("Task not found with ID: " + id));
+    }
+
+    @Override
+    @ExecutionTime
+    @Transactional(readOnly = true)
+    public TaskResponseDto getTaskById(UUID id) {
+        return taskRepository.findById(id)
+                .map(taskMapper::toTaskResponseDto)
+                .orElseThrow(() -> new TaskNotFoundException("Task not found with ID: " + id));
+    }
+
+    @Override
+    @ExecutionTime
+    @Transactional(readOnly = true)
+    public Page<TaskResponseDto> getTasksByAuthor(UUID authorId, Pageable pageable) {
+        Page<Task> tasks = taskRepository.findAllByAuthor_Id(authorId, pageable);
+        tasks.forEach(this::loadTaskAssociations);
+
+        return tasks.map(taskMapper::toTaskResponseDto);
+    }
+
+    @Override
+    @ExecutionTime
+    @Transactional(readOnly = true)
+    public Page<TaskResponseDto> getTasksByAssignee(UUID assigneeId, Pageable pageable) {
+        Page<Task> tasks = taskRepository.findAllByTaskAssignees_Assignee_Id(assigneeId, pageable);
+        tasks.forEach(this::loadTaskAssociations);
+
+        return tasks.map(taskMapper::toTaskResponseDto);
+
+    }
+
+    @Override
+    @ExecutionTime
+    @Transactional(readOnly = true)
+    public Page<TaskResponseDto> getAllTasks(Pageable pageable) {
+        Page<Task> tasks = taskRepository.findAllWithDetails(pageable);
+        tasks.forEach(this::loadTaskAssociations);
+
+        return tasks.map(taskMapper::toTaskResponseDto);
+    }
+
+    @Override
+    @Transactional
+    public TaskResponseDto createTask(CreateTaskRequestDto request) {
+        return Optional.of(request)
+                .map(taskMapper::toTaskForCreate)
+                .map(taskRepository::save)
+                .map(taskMapper::toTaskResponseDto)
+                .orElseThrow(() -> new EntitySaveException("Failed to save task."));
+    }
+
+
+    @Override
+    @Transactional
+    public void deleteTask(UUID id) {
+        Task task = taskRepository.findById(id)
+                .orElseThrow(() -> new TaskNotFoundException("Task not found with ID: " + id));
+
+        taskRepository.delete(task);
+    }
+
+    @Override
+    @Transactional
+    public void save(Task task) {
+        taskRepository.save(task);
+    }
```

Comment:
why u need create and save methods?) it s not easy to understand why ur service have 2 methods do same things